sqlmesh/core/metric/definition.py

```diff
@@ -181,10 +185,40 @@ def _raise_metric_config_error(msg: str, path: Path) -> None:
     raise ConfigError(f"{msg}. '{path}'")
 
 
-def _find_tables(expression: exp.Expression, dialect: str) -> t.Set[str]:
-    """Finds all the table references in a metric definition."""
-    return {
-        d.normalize_model_name(exp.table_(*reversed(column.parts[:-1])), dialect=dialect)  # type: ignore
-        for column in expression.find_all(exp.Column)
-        if column.table
-    }
+def _get_ordered_tables(expression: exp.Expression, dialect: str) -> t.List[str]:
+    """Finds all the table references in a metric definition.
+
+    Additionally ensure than the first table returned is the 'measure' or numeric value being aggregated.
```

Comment:
Can there only be a single measure table? Should we just express this in a return value as tuple (preferably a named one / pydantic model)